src/ja/nicovideoseiga/src/eu/kanade/tachiyomi/extension/ja/nicovideoseiga/NicovideoSeiga.kt

```diff
@@ -1,216 +1,183 @@
-                        baseUrl + chapter.select("div > div.description > div.title > a")
-                            .attr("href"),
-                    )
-                    // The data-number attribute is the only way we can determine chapter orders,
-                    // without that this extension would have been impossible to make
-                    // Note: Promotional materials also count as "chapters" here, so auto tracking unfortunately does not work at all
-                    chapter_number = chapter.select("div").attr("data-number").toFloat()
-                    // We can't determine the upload date from the website
-                    // Store date_upload when a chapter is found for the first time
-                    val dateFound = System.currentTimeMillis()
-                    if (!sharedPref.contains(chapter_number.toString())) {
-                        editor.putLong(chapter_number.toString(), dateFound)
-                    }
-                    date_upload = sharedPref.getLong(chapter_number.toString(), dateFound)
+                    name = (if (isPaid) "\uD83D\uDCB4 " else "") + chapter.meta.title
+                    // Timestamp is in seconds, convert to milliseconds
+                    date_upload = chapter.meta.createdAt * 1000
+                    // While chapters are properly sorted, authors often add promotional material as "chapters" which breaks trackers
+                    // There's no way to properly filter these as they are treated the same as normal chapters
+                    chapter_number = chapter.meta.number.toFloat()
+                    // Can't use setUrlWithoutDomain as it uses the baseUrl instead of apiUrl
+                    url = "/episodes/${chapter.id}/frames"
                 },
             )
         }
-        editor.apply()
         chapters.sortByDescending { chapter -> chapter.chapter_number }
         return chapters
     }
 
+    override fun chapterListRequest(manga: SManga): Request {
+        // Overwrite to use the API instead of scraping the shared URL
+        val id = manga.url.substringAfterLast("/")
+        return GET("$apiUrl/contents/$id/episodes")
+    }
+
+    override fun getChapterUrl(chapter: SChapter): String {
+        return "$baseUrl/watch/mg${chapter.url.substringBeforeLast("/").substringAfterLast("/")}"
+    }
+
     override fun pageListParse(response: Response): List<Page> {
-        val doc = response.asJsoup()
-        val pages = ArrayList<Page>()
-        // Nicovideo will refuse to serve any pages if the user has not logged in
-        if (!doc.select("#login_manga").isEmpty()) {
-            throw SecurityException("Not logged in. Please login via WebView first")
+        // Nicomanga historically refuses to serve pages if you don't login.
+        // However, due to the network attack against the site (as of July 2024) login is disabled
+        // Changes may be required as the site recovers
+        if (response.code == 403) {
```

Comment:
app will check response code and when not 2xx, throw an http exception, before reaching this function.

you can override `fetchPageList` to manually check response codes and throw necessary exception or throw exception in an interceptor, [example](https://github.com/keiyoushi/extensions-source/blob/ddea522ab65be1b0a2ff29af04e6114833a2e6b9/src/ja/comicfuz/src/eu/kanade/tachiyomi/extension/ja/comicfuz/ComicFuz.kt#L36-L50)